mllib/src/test/scala/org/apache/spark/ml/classification/DecisionTreeClassifierSuite.scala

```diff
@@ -385,6 +385,20 @@ class DecisionTreeClassifierSuite
     testEstimatorAndModelReadWrite(dt, continuousData, allParamSettings ++ Map("maxDepth" -> 0),
       allParamSettings ++ Map("maxDepth" -> 0), checkModelData)
   }
+
+  test("read/write: ImpurityCalculator builder did not recognize impurity type: Gini") {
+    val rdd = TreeTests.getTreeReadWriteData(sc)
+
+    val categoricalData: DataFrame =
+      TreeTests.setMetadata(rdd, Map(0 -> 2, 1 -> 3), numClasses = 2)
```

Comment:
To simplify this, you can write ```TreeTests.setMetadata(rdd, Map.empty[Int, Int], numClasses = 2)```.
Since this is not testing categorical features, there's no need to throw them in.